prowler/providers/azure/azure_provider.py

```diff
@@ -503,8 +576,10 @@ def check_service_principal_creds_env_vars():
                 logger.critical(
                     f"Azure provider: Missing environment variable {env_var} needed to authenticate against Azure"
                 )
-                # TODO: add custom exception once we have the AzureException
-                raise SystemExit
+                raise AzureEnvironmentVariableError(
+                    file=os.path.basename(__file__),
+                    message=f"Missing environment variable {env_var}",
```

Comment:
I think we should have the same message as the logger, something like:
`Missing environment variable {env_var} required to authenticate`.

Reply:
Done